test/CMakeLists.txt

```diff
@@ -1,7 +1,23 @@
+add_subdirectory(external/catch)
 add_subdirectory(util)
 add_custom_target(benchmarks)
 add_subdirectory(object-store)
 
+if(CMAKE_SYSTEM_NAME MATCHES "^Windows")
+    # Increase the Catch2 virtual console width because our test names can be very long and they break test reports
+    set(CATCH_CONFIG_CONSOLE_WIDTH 300)
+elseif(APPLE AND NOT CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+    set(CATCH_CONFIG_NO_POSIX_SIGNALS ON CACHE BOOL "Disable Catch support for POSIX signals on Apple platforms other than macOS")
+endif()
+set(CATCH_CONFIG_ENABLE_ALL_STRINGMAKERS ON CACHE BOOL "Enable Catch2's optional standard library stringmakers")
```

Comment:
The `CATCH_CONFIG_` variables need to be set before including Catch itself, I believe.